react-events-and-state/src/index.jsx

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+class CustomButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { isClicked: false };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick() {
+    this.setState(prevState => ({
+      buttonClicked: !prevState.isClicked
+    }));
+  }
+
+  render() {
+    return <button onClick={this.handleClick}>
+      {this.state.buttonClicked ? 'Thanks!' : 'Click me!'}
```

Comment:
Instead of adding a new `buttonClicked` property to your state, just update the `isClicked` property.

```js
    this.setState(prevState => ({
      isClicked: !prevState.isClicked
    }));
```